main.c

```diff
@@ -26,21 +26,37 @@ bool box_intersect(const struct slurp_box *a, const struct slurp_box *b) {
 		a->height + a->y > b->y;
 }
 
+struct slurp_selection *seat_get_current_selection(struct slurp_seat *seat) {
+	if (seat->pointer_selection.has_selection) {
+		return &seat->pointer_selection;
+	} else if (seat->touch_selection.has_selection) {
+		return &seat->touch_selection;
+	} else {
+		return NULL;
+	}
+}
+
 static struct slurp_output *output_from_surface(struct slurp_state *state,
 	struct wl_surface *surface);
 
 static void move_seat(struct slurp_seat *seat, wl_fixed_t surface_x,
```

Comment:
This function should probably take a `struct slurp_selection *`. When called from the pointer event handler, it'd be called with `pointer_selection`. When called from the touch event handler, it'd be called with `touch_selection`.

Reply:
It's also called from move_seat and render, besides its whole point is to determine which selection is active right now.

Reply:
b3be2ebf17a959e59ea449e643ee953bace8a583

On second thought I've just added the slurp_selection argument to the relevant functions (move_seat, handle_selection*) and made a simple check in render as it was the only place left. seat_get_current_selection is thus unneeded and all the crutches are gone.

Reply:
Nice!